python/pyarrow/tests/test_types.py

```diff
@@ -533,3 +548,38 @@ def test_schema_from_pandas(data):
     schema = pa.Schema.from_pandas(df)
     expected = pa.Table.from_pandas(df).schema
     assert schema == expected
+
+
+@h.given(
+    past.all_types |
+    past.all_fields |
+    past.all_schemas
+)
```

Comment:
What sort of exhaustiveness guarantees does hypothesis provide? I guess I should check this out and run the tests myself =)

Reply:
It's pretty cool, slowish but We can improve that later.

Reply:
Try: `pytest pyarrow/tests/test_types.py -sv --only-hypothesis --hypothesis-profile=debug`

Reply:
My point is that we should have a good understanding of what percentage of the sample space is covered by the tests. Is it 5%, 1%, 0.1%, 0.01%? 

In the actual data generation I would not leave the data generation to Hypothesis =)